File: lib/src/main/java/com/asyncanimator/launcher/pending/NullableAnimatorListener.java
package com.asyncanimator.launcher.pending;

import com.asyncanimator.core.anim.Animator;

/**
 * NullableAnimatorListener — Animator 参数可空的 listener 接口。
 *
 * <p>对应分析文档 §6.3.4。
 */
public interface NullableAnimatorListener {
    default void onAnimationCancel(Animator animator) {}
    default void onAnimationEnd(Animator animator) {}
    default void onAnimationStart(Animator animator) {}
}

File: lib/src/main/java/com/asyncanimator/launcher/pending/NullableAnimatorListener.kt
package com.asyncanimator.launcher.pending

import android.animation.Animator

/**
 * NullableAnimatorListener — Animator listener 接口（历史上演进自"参数可空"）。
 *
 * 对应分析文档 §6.3.4。换平台 `android.animation.Animator` 后参数为 `@NonNull`，
 * 现派发真实 animator，可空容忍仅作历史注释保留。
 */
interface NullableAnimatorListener {
    fun onAnimationCancel(animator: Animator) {}
    fun onAnimationEnd(animator: Animator) {}
    fun onAnimationStart(animator: Animator) {}
}
